app/routes.py

```diff
@@ -351,54 +351,100 @@ def add_harvest_source():
     )
 
 
-# View Source
+@mod.route("/harvest_source/<source_id>", methods=["GET"])
+def view_harvest_source_data(source_id: str):
+    source = db.get_harvest_source(source_id)
+    jobs = db.get_all_harvest_jobs_by_filter(
+        {"harvest_source_id": source.id, "status": "complete"}
+    )
+    next_job = "N/A"
+    future_jobs = db.get_new_harvest_jobs_by_source_in_future(source.id)
+    if len(future_jobs):
+        next_job = future_jobs[0].date_created
+    chartdata = {
+        "labels": [job.date_finished for job in jobs],
+        "datasets": [
+            {
+                "label": "Added",
+                "data": [job.records_added for job in jobs],
+                "borderColor": "green",
+                "backgroundColor": "green",
+            },
+            {
+                "label": "Deleted",
+                "data": [job.records_deleted for job in jobs],
+                "borderColor": "black",
+                "backgroundColor": "black",
+            },
+            {
+                "label": "Errored",
+                "data": [job.records_errored for job in jobs],
+                "borderColor": "red",
+                "backgroundColor": "red",
+            },
+            {
+                "label": "Igrored",
```

Comment:
typo. not trying to nitpick just wanna make sure your stuff gets looked at!

Reply:
good catch!